centreon/src/Core/Security/Vault/Application/UseCase/UpdateVaultConfiguration/NewVaultConfigurationFactory.php

```diff
@@ -50,11 +50,11 @@ public function __construct(private readonly EncryptionInterface $encryption) {
      *
      * @return NewVaultConfiguration
      */
-    public function create(CreateVaultConfigurationRequest $request): NewVaultConfiguration
+    public function create(UpdateVaultConfigurationRequest $request): NewVaultConfiguration
     {
         return new NewVaultConfiguration(
             $this->encryption,
-            $request->name,
+            'hashicorp_vault',
```

Comment:
use a constant in NewVaultConfiguration/VaultConfiguration to avoid magic string